src/scene/graph-node.js

```diff
@@ -519,7 +518,7 @@ class GraphNode extends EventHandler {
             for (let i = 0; i < len; ++i) {
                 const descendants = this._children[i].find(attr, value);
                 if (descendants.length)
-                    results = results.concat(descendants);
+                    results.push(...descendants);
```

Comment:
I could imagine that `descendants.length` is only checked for optimisation to not create a new results array all the time, but since `push` in the case of zero length would just do nothing, there shouldn't be any need for length checking anymore?

Reply:
That's right. I modified those parts and also made other changes such as moving parent/children recursion out of if...else blocks so they're not duplicated anymore.